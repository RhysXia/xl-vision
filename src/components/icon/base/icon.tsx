import classnames from 'classnames'
import * as React from 'react'
import { clsPrefix } from '../../_commons/config'

export interface IconProps {
    className?: string
    style?: React.CSSProperties
    children: React.ReactSVGElement
    onClick?: React.MouseEventHandler<HTMLElement>
}

const iconClsPrefix = `${clsPrefix}-icon`

const Icon: React.FunctionComponent<IconProps> = props => {
    const { className, ...others } = props
    const classes = classnames([
        iconClsPrefix
    ], className)
    return (
        <i className={classes} {...others}/>
    )
}

export default Icon
