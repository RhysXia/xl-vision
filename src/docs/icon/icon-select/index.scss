@use '~site/assets/style/var' as *;

.icon-select {
  width: 100%;
  padding: 1rem 0;
  text-align: center;

  .icon-input {
    display: inline-block;
    width: 70%;
    padding: 0.5rem;
    color: $text-color;
    font-size: 1.2rem;
    line-height: 1.5;
    border: 1px $border-color solid;
    border-radius: $border-radius;
    outline: none;
    transition: border-color 0.4s ease-in-out;

    &:hover {
      border-color: darken($border-color, 0.1);
    }

    &:focus {
      border-color: darken($border-color, 0.1);
    }
  }

  .icon-info {
    padding: 1rem 0;
    color: $text-color;
  }

  .icon-list {
    display: inline-block;
    width: 100%;
  }

  .icon-wrapper {
    position: relative;
    display: inline-block;
    box-sizing: border-box;
    width: 200px;
    margin: 1rem 0;
    text-align: center;

    .icon {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      box-sizing: border-box;
      width: 100%;
      height: 120px;
      border-radius: $border-radius;
      cursor: pointer;
      transition: background-color 0.4s ease-in-out;

      svg {
        transition: fill 0.4s ease-in-out;
      }

      &:hover {
        background-color: $primary-color;

        svg {
          fill: #fff;
        }
      }
    }

    .icon-name {
      position: absolute;
      bottom: 8px;
      left: 50%;
      padding-top: 0.5rem;
      color: #fff;
      font-size: 0.8rem;
      transform: translateX(-50%);
      transition: opacity 0.4s ease-in-out;
    }
  }
}
