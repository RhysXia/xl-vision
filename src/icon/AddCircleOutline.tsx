/* eslint-disable */
// This file is automatically generated

import React from 'react'
import createIcon from './base/createIcon'

const svgElement = (
    <svg viewBox='0 0 512 512'>
        <path fill='none' strokeMiterlimit='10' strokeWidth='32' d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z'/><path fill='none' strokeLinecap='round' strokeLinejoin='round' strokeWidth='32' d='M256 176L256 336'/><path fill='none' strokeLinecap='round' strokeLinejoin='round' strokeWidth='32' d='M336 256L176 256'/>
    </svg>
)

const AddCircleOutline = createIcon(svgElement)

AddCircleOutline.displayName = 'AddCircleOutline'

export { IconProps } from './base/createIcon'

export default AddCircleOutline
