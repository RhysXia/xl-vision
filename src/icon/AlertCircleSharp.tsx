/* eslint-disable */
// This file is automatically generated

import React from 'react'
import createIcon from './base/createIcon'

const svgElement = (
    <svg viewBox='0 0 512 512'>
        <path fill='none' d='M240 304L272 304 278 144 234 144 240 304z'/><path d='M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm20 319.91h-40v-40h40zM272 304h-32l-6-160h44z'/>
    </svg>
)

const AlertCircleSharp = createIcon(svgElement)

AlertCircleSharp.displayName = 'AlertCircleSharp'

export { IconProps } from './base/createIcon'

export default AlertCircleSharp
