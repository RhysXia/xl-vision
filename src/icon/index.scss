@import '../style/commons/var';

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

.#{$--cls-prefix}-icon {
  display: inline-block;
  font-size: $--icon-font-size;
  line-height: 0;

  &--spin {
    animation: spin $--icon-spin-time linear infinite;
  }
}
