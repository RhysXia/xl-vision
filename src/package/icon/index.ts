// This file is automatically generated
// tslint:disabled

import Icon from './base/baseIcon'
import FaBr500px from './icons/FaBr500px'

const IconObj: any = Icon
IconObj.FaBr500px = FaBr500px

export default IconObj
