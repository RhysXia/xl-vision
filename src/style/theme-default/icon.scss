@import './_commons/var.scss';

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
}

.#{$--cls-prefix}-icon {
    font-size: 1rem;
    display: inline-block;
    line-height: 0;
    &--spin{
        animation: spin .4s linear infinite;
    }
}