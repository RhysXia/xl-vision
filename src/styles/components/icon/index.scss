@use '../../theme' as *;

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}

.#{$cls-prefix}-icon {
  display: inline-block;
  line-height: 0;
  text-align: center;
  vertical-align: -0.125em; // for SVG icon, see https://blog.prototypr.io/align-svg-icons-to-text-and-say-goodbye-to-font-icons-d44b3d7b26b4
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  &--spin {
    animation: spin $icon-spin-duration linear infinite;
  }

  > * {
    line-height: 1;
  }

  svg {
    display: inline-block;
  }
}
