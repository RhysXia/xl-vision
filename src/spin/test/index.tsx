import { mount } from 'enzyme'
import * as React from 'react'
import Spin from '..'
import { FasCircleNotch } from '../../icon'

describe('spin', () => {
  it('基本用法', () => {
    const wrapper = mount(<Spin/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('内嵌内容', () => {
    const wrapper = mount(
      <Spin>
        <div className='inner'>Further details about the context of this alert.</div>
      </Spin>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('覆盖父元素', () => {
    const wrapper = mount(
      <div className='inner'>
        Further details about the context of this alert.
        <Spin cover={true}/>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('不同大小', () => {
    const wrapper = mount(
      <div className='wrapper'>
        <Spin size='small'/>
        <Spin size='default'/>
        <Spin size='large'/>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('自定义描述', () => {
    const wrapper = mount(
      <Spin tip='Loading...'/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('自定义指示器', () => {
    const wrapper = mount(
      <Spin indicator={<FasCircleNotch spin={true}/>} tip='Loading...'/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test.todo('启动加载和关闭加载')
  // it('启动加载和关闭加载', async () => {
  //   const TestComp = () => {
  //     const [display, setDisplay] = React.useState(true)
  //     const clickHandler = () => {
  //       setDisplay(!display)
  //     }
  //     return (
  //       <div className='wrapper'>
  //         <Spin spinning={display}>
  //           <div className='inner'>Further details about the context of this alert.</div>
  //         </Spin>
  //         <Button onClick={clickHandler}>{display ? '关闭' : '打开'}</Button>
  //       </div>
  //     )
  //   }
  //   const wrapper = mount(
  //     <TestComp/>
  //   )
  //   expect(wrapper.find('.xl-spin--spinning')).not.toBeNull()
  //   wrapper.find('button').simulate('click')
  //   await sleep(500)
  //   console.log(wrapper.find('.xl-spin--spinning').debug())
  //   expect(wrapper.find('.xl-spin--spinning')).toBeNull()
  // })
})

// function sleep (time: number) {
//   return new Promise(resolve => {
//     setTimeout(() => {
//       resolve()
//     }, time)
//   })
// }
