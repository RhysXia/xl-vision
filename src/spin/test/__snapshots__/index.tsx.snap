// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spin 不同大小 1`] = `
<div
  className="wrapper"
>
  <xl-spin
    size="small"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <xl-css-transition
        classNames="xl-spin__fade"
        in={true}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <xl-transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          in={true}
          leave={[Function]}
          leaveCancelled={[Function]}
          mountOnEnter={true}
          unmountOnLeave={true}
        >
          <div
            className="xl-spin__fade"
          >
            <div
              className="xl-spin__wrapper xl-spin__wrapper--small"
            >
              <span
                className="xl-spin__indicator"
              >
                <xl-fas-spinner
                  spin={true}
                >
                  <xl-icon
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </xl-icon>
                </xl-fas-spinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </xl-transition>
      </xl-css-transition>
    </div>
  </xl-spin>
  <xl-spin
    size="default"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <xl-css-transition
        classNames="xl-spin__fade"
        in={true}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <xl-transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          in={true}
          leave={[Function]}
          leaveCancelled={[Function]}
          mountOnEnter={true}
          unmountOnLeave={true}
        >
          <div
            className="xl-spin__fade"
          >
            <div
              className="xl-spin__wrapper xl-spin__wrapper--default"
            >
              <span
                className="xl-spin__indicator"
              >
                <xl-fas-spinner
                  spin={true}
                >
                  <xl-icon
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </xl-icon>
                </xl-fas-spinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </xl-transition>
      </xl-css-transition>
    </div>
  </xl-spin>
  <xl-spin
    size="large"
  >
    <div
      className="xl-spin xl-spin--spinning"
    >
      <xl-css-transition
        classNames="xl-spin__fade"
        in={true}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <xl-transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          in={true}
          leave={[Function]}
          leaveCancelled={[Function]}
          mountOnEnter={true}
          unmountOnLeave={true}
        >
          <div
            className="xl-spin__fade"
          >
            <div
              className="xl-spin__wrapper xl-spin__wrapper--large"
            >
              <span
                className="xl-spin__indicator"
              >
                <xl-fas-spinner
                  spin={true}
                >
                  <xl-icon
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </xl-icon>
                </xl-fas-spinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </xl-transition>
      </xl-css-transition>
    </div>
  </xl-spin>
</div>
`;

exports[`spin 内嵌内容 1`] = `
<xl-spin>
  <div
    className="xl-spin xl-spin--spinning xl-spin--nested"
  >
    <xl-css-transition
      classNames="xl-spin__fade"
      in={true}
      mountOnEnter={true}
      unmountOnLeave={true}
    >
      <xl-transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        in={true}
        leave={[Function]}
        leaveCancelled={[Function]}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <div
          className="xl-spin__fade"
        >
          <div
            className="xl-spin__wrapper xl-spin__wrapper--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <xl-fas-spinner
                spin={true}
              >
                <xl-icon
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </xl-icon>
              </xl-fas-spinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              loading
            </span>
          </div>
        </div>
      </xl-transition>
    </xl-css-transition>
    <div
      className="xl-spin__children"
    >
      <div
        className="inner"
      >
        Further details about the context of this alert.
      </div>
    </div>
  </div>
</xl-spin>
`;

exports[`spin 基本用法 1`] = `
<xl-spin>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <xl-css-transition
      classNames="xl-spin__fade"
      in={true}
      mountOnEnter={true}
      unmountOnLeave={true}
    >
      <xl-transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        in={true}
        leave={[Function]}
        leaveCancelled={[Function]}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <div
          className="xl-spin__fade"
        >
          <div
            className="xl-spin__wrapper xl-spin__wrapper--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <xl-fas-spinner
                spin={true}
              >
                <xl-icon
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </xl-icon>
              </xl-fas-spinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              loading
            </span>
          </div>
        </div>
      </xl-transition>
    </xl-css-transition>
  </div>
</xl-spin>
`;

exports[`spin 自定义指示器 1`] = `
<xl-spin
  indicator={
    <xl-fas-circle-notch
      spin={true}
    />
  }
  tip="Loading..."
>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <xl-css-transition
      classNames="xl-spin__fade"
      in={true}
      mountOnEnter={true}
      unmountOnLeave={true}
    >
      <xl-transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        in={true}
        leave={[Function]}
        leaveCancelled={[Function]}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <div
          className="xl-spin__fade"
        >
          <div
            className="xl-spin__wrapper xl-spin__wrapper--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <xl-fas-circle-notch
                spin={true}
              >
                <xl-icon
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
                      />
                    </svg>
                  </i>
                </xl-icon>
              </xl-fas-circle-notch>
            </span>
            <span
              className="xl-spin__tip"
            >
              Loading...
            </span>
          </div>
        </div>
      </xl-transition>
    </xl-css-transition>
  </div>
</xl-spin>
`;

exports[`spin 自定义描述 1`] = `
<xl-spin
  tip="Loading..."
>
  <div
    className="xl-spin xl-spin--spinning"
  >
    <xl-css-transition
      classNames="xl-spin__fade"
      in={true}
      mountOnEnter={true}
      unmountOnLeave={true}
    >
      <xl-transition
        afterAppear={[Function]}
        afterEnter={[Function]}
        afterLeave={[Function]}
        appear={[Function]}
        appearCancelled={[Function]}
        beforeAppear={[Function]}
        beforeEnter={[Function]}
        beforeLeave={[Function]}
        enter={[Function]}
        enterCancelled={[Function]}
        in={true}
        leave={[Function]}
        leaveCancelled={[Function]}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <div
          className="xl-spin__fade"
        >
          <div
            className="xl-spin__wrapper xl-spin__wrapper--default"
          >
            <span
              className="xl-spin__indicator"
            >
              <xl-fas-spinner
                spin={true}
              >
                <xl-icon
                  spin={true}
                >
                  <i
                    className="xl-icon xl-icon--spin"
                    style={Object {}}
                  >
                    <svg
                      fill="currentColor"
                      style={
                        Object {
                          "height": "1em",
                          "width": "1em",
                        }
                      }
                      viewBox="0 0 512 512"
                    >
                      <path
                        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                      />
                    </svg>
                  </i>
                </xl-icon>
              </xl-fas-spinner>
            </span>
            <span
              className="xl-spin__tip"
            >
              Loading...
            </span>
          </div>
        </div>
      </xl-transition>
    </xl-css-transition>
  </div>
</xl-spin>
`;

exports[`spin 覆盖父元素 1`] = `
<div
  className="inner"
>
  Further details about the context of this alert.
  <xl-spin
    cover={true}
  >
    <div
      className="xl-spin xl-spin--spinning xl-spin--cover"
    >
      <xl-css-transition
        classNames="xl-spin__fade"
        in={true}
        mountOnEnter={true}
        unmountOnLeave={true}
      >
        <xl-transition
          afterAppear={[Function]}
          afterEnter={[Function]}
          afterLeave={[Function]}
          appear={[Function]}
          appearCancelled={[Function]}
          beforeAppear={[Function]}
          beforeEnter={[Function]}
          beforeLeave={[Function]}
          enter={[Function]}
          enterCancelled={[Function]}
          in={true}
          leave={[Function]}
          leaveCancelled={[Function]}
          mountOnEnter={true}
          unmountOnLeave={true}
        >
          <div
            className="xl-spin__fade"
          >
            <div
              className="xl-spin__wrapper xl-spin__wrapper--default"
            >
              <span
                className="xl-spin__indicator"
              >
                <xl-fas-spinner
                  spin={true}
                >
                  <xl-icon
                    spin={true}
                  >
                    <i
                      className="xl-icon xl-icon--spin"
                      style={Object {}}
                    >
                      <svg
                        fill="currentColor"
                        style={
                          Object {
                            "height": "1em",
                            "width": "1em",
                          }
                        }
                        viewBox="0 0 512 512"
                      >
                        <path
                          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
                        />
                      </svg>
                    </i>
                  </xl-icon>
                </xl-fas-spinner>
              </span>
              <span
                className="xl-spin__tip"
              >
                loading
              </span>
            </div>
          </div>
        </xl-transition>
      </xl-css-transition>
    </div>
  </xl-spin>
</div>
`;
